include/seqan3/search/configuration/on_result.hpp

```diff
@@ -0,0 +1,88 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::search_cfg::on_result.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <type_traits>
+
+#include <range/v3/utility/semiregular_box.hpp>
+
+#include <seqan3/core/algorithm/pipeable_config_element.hpp>
+#include <seqan3/search/configuration/detail.hpp>
+
+namespace seqan3::search_cfg
+{
+
+/*!\brief Configuration element to provide a user defined callback function for the search.
+ * \ingroup search_configuration
+ *
+ * \tparam callback_t The type of the callback; must model std::invocable with the generated seqan3::search_result
```

Comment:
```suggestion
 * \tparam callback_t The type of the callback; must model std::invocable with the generated seqan3::search_result
```

Btw, is the invocable part checked somewhere (not in the config itself obviously); or will it just fail not so graciously?

Reply:
yes it is checked in the execution handlers and in the function call operator of the search algorithms.